typegen/strats/inline.py

```diff
@@ -301,6 +301,11 @@ def leave_FunctionDef(
 
         rettypes = self._get_trace_for_rettype(original_node)
 
+        if updated_node.returns is not None:
```

Comment:
This check can be performed prior to the assignment to `rettypes` in the line above

Reply:
Done

Reply:
Done